server/server.go

```diff
@@ -258,11 +261,17 @@ func New(cfg Config) (*Server, error) {
 		httpServer.TLSConfig = httpTLSConfig
 	}
 
+	var handler *signals.Handler
+	if !cfg.DisableSignalHandling {
+		handler = signals.NewHandler(log)
+	}
+
 	return &Server{
 		cfg:          cfg,
 		httpListener: httpListener,
 		grpcListener: grpcListener,
-		handler:      signals.NewHandler(log),
+		handler:      handler,
+		quit:         make(chan struct{}),
```

Comment:
Would it be better to make the changes to `signals.Handler` with a new method like `handler.Watch()`? (or possibly a better name 😄)

Reply:
I'm not sure sure that signal handler that doesn't handle signals is a way to go.